src/reflect/scala/reflect/internal/ReificationSupport.scala

```diff
@@ -611,7 +611,7 @@ trait ReificationSupport { self: SymbolTable =>
     def UnliftListOfListsElementwise[T](unliftable: Unliftable[T]) = new UnliftListOfListsElementwise[T] {
       def unapply(lst: List[List[Tree]]): Option[List[List[T]]] = {
         val unlifted = lst.map { l => l.flatMap { unliftable.unapply(_) } }
-        if (unlifted.flatten.length == lst.flatten.length) Some(unlifted) else None
+        if (sumSize(unlifted, 0) == sumSize(lst, 0)) Some(unlifted) else None
```

Comment:
This is a weird test... isn't it really "did all of this unlifts succeed"? Not sure if it could be made more elegant, but there may be a more efficient way to check it.

Reply:
Looking at it, and think there is... the nature of the loop is that, if it succeeds, not only would the sums of lengths be equal, but every inner list would have the same length as its source.